### PR TITLE
Updated cimport for numpy to maintain cimport consistency

### LINKED ISSUE #3908: change `cimport numpy as np` to `cimport numpy as cnp` for consistency
## Is your feature request related to a problem? ##

We should be consistent with the way we cimport numpy in the codebase. 

Lots of places currently use
```cython
 cimport numpy as np
```

While the more modern approach and the one taken in some PRs #3777 , `timestep.pyx` and a few random other existing places is to

```cython 
 cimport numpy as cnp
```

Personally I prefer the second one (and its the one used by pandas) as it is much clearer when the C API is being used vs python API. 
@tylerjereddy  do you know if there is a standard for scipy?

Regardless of which way we go we should be consistent if possible.

Let me know what you think on this issue and if we come to a conclusion I can raise a seperate issue for taking some action. 
Might be a nice easy `good-first-issue` etc.

### PR DESCRIPTION
**Pull request by Sumit112192**
Fixes #3908 

Changes made in this Pull Request:
 - Have updated all `cimport numpy` from `np` to `cnp`

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

## Developers certificate of origin
- [x] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

----
:books: Documentation preview :books:: https://mdanalysis--4346.org.readthedocs.build/en/4346/

### CODE REVIEW

**Comment by RMeli** [comment]
Thanks for the PR @Sumit112192. Looks like a few tests timed out, I'll restart them before having a deeper look.

**Comment by RMeli** [comment]
In the meantime, please add yourself to the `AUTHORS` file.

**Comment by IAlibay** [comment]
I believe part of the work that needs doing is checking (maybe adding) benchmarks to ensure that performance isn't affected? If I remember correctly that was an ask from @hmacdope last time someone started a PR on this.

**Comment by RMeli** [comment]
Indeed: https://github.com/MDAnalysis/mdanalysis/pull/3978#issuecomment-1369255654

**Comment by hmacdope** [comment]
@Sumit112192 @IAlibay @RMeli Yes, thanks for taking an intial look here. The issue is that there is a difference in speed calling the C vs normal numpy api, and there are some cases where the C API is less appropriate. Previously we used aliasing which was unclear, but we just need to run the benchmarks to make sure there are no major performance regressions. The way to do this is https://github.com/MDAnalysis/mdanalysis/blob/develop/benchmarks/README.rst. Something like `asv continuous <commit on main> <commit at HEAD of this PR >`

**Comment by hmacdope** [comment]
I imagine if anything performance will improve but we should check

**Comment by Sumit112192** [comment]
@hmacdope, I am getting an issue with asv.conf.json. In that file, it mentions using python 3.8 version. But in env/09b6664d368df460776d8e73bd3171a0/project/package/pyproject.toml it mentions to use python 3.9+

I ran the command `asv continuous 433603 f4005d -e --verbose` and got some error regarding python versions and I believe the error was related to the above discrepancy in python versions.

The error
```
`(my-project-env) sam@sam-Latitude-5480:~/mdanalysis/benchmarks$ asv continuous 433603 f4005d -e --verbose
· Running '/usr/bin/git config init.defaultBranch'
· Running '/usr/bin/git config --global init.defaultBranch'
· Running '/usr/bin/git config --system init.defaultBranch'
· Running '/usr/bin/git rev-parse f4005d^{commit}'
  OUTPUT -------->
  f4005db383f3f4c446433fb32156236f29f251ba
· Running '/usr/bin/git rev-parse 433603^{commit}'
  OUTPUT -------->
  43360341b56823ccc2459a0cf9cc709c2b30283f
· Running '/usr/bin/git config init.defaultBranch'
· Running '/usr/bin/git config --global init.defaultBranch'
· Running '/usr/bin/git config --system init.defaultBranch'
· Creating environments
·· Running '/home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/bin/python -c pass'
· Discovering benchmarks
·· Running '/usr/bin/git rev-parse develop^{commit}'
   OUTPUT -------->
   f4005db383f3f4c446433fb32156236f29f251ba
·· Running '/usr/bin/git -c protocol.file.allow=always submodule deinit -f .'
·· Running '/usr/bin/git checkout -f f4005db383f3f4c446433fb32156236f29f251ba'
   OUTPUT -------->
   Previous HEAD position was 43360341b Updated cimport for numpy to maintain cimport consistency
   HEAD is now at f4005db38 Expose C distance backend in libmdanalysis.pxd (#4342)
·· Running '/usr/bin/git clean -fdx'
·· Running '/usr/bin/git -c protocol.file.allow=always submodule update --init --recursive'
·· Uninstalling from conda-py3.8-Cython-MDAnalysisTests-mock-nose-numpy-pytest-scipy-six
·· Running '/home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/bin/python -mpip uninstall -y mdanalysis'
   OUTPUT -------->
   WARNING: Skipping mdanalysis as it is not installed.
·· Running '/usr/bin/git name-rev --name-only --exclude=remotes/* --no-undefined f4005db383f3f4c446433fb32156236f29f251ba'
   OUTPUT -------->
   develop
·· Building f4005db3 <develop> for conda-py3.8-Cython-MDAnalysisTests-mock-nose-numpy-pytest-scipy-six
·· Running '/home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/bin/python -mpip wheel --no-deps --no-index -w /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/asv-build-cache/f4005db383f3f4c446433fb32156236f29f251ba /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/project/package'
   OUTPUT -------->
   Processing /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/project/package
     Preparing metadata (pyproject.toml): started
     Preparing metadata (pyproject.toml): finished with status 'error'
     error: subprocess-exited-with-error
     
     × Preparing metadata (pyproject.toml) did not run successfully.
     │ exit code: 255
     ╰─> [2 lines of output]
         MDAnalysis requires Python 3.9+. Python 3.8 detected
         Please upgrade your version of Python.
         [end of output]
   
     note: This error originates from a subprocess, and is likely not a problem with pip.
   error: metadata-generation-failed

   
   × Encountered error while generating package metadata.
   ╰─> See above for output.
   
   note: This is an issue with the package mentioned above, not pip.
   hint: See above for details.

·· Error running /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/bin/python -mpip wheel --no-deps --no-index -w /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/asv-build-cache/f4005db383f3f4c446433fb32156236f29f251ba /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/project/package (exit status 1)
·· Failed: trying different commit/environment
·· Running '/usr/bin/git -c protocol.file.allow=always submodule deinit -f .'
·· Running '/usr/bin/git checkout -f 43360341b56823ccc2459a0cf9cc709c2b30283f'
   OUTPUT -------->
   Previous HEAD position was f4005db38 Expose C distance backend in libmdanalysis.pxd (#4342)
   HEAD is now at 43360341b Updated cimport for numpy to maintain cimport consistency
·· Running '/usr/bin/git clean -fdx'
·· Running '/usr/bin/git -c protocol.file.allow=always submodule update --init --recursive'
·· Uninstalling from conda-py3.8-Cython-MDAnalysisTests-mock-nose-numpy-pytest-scipy-six
·· Running '/home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/bin/python -mpip uninstall -y mdanalysis'
   OUTPUT -------->
   WARNING: Skipping mdanalysis as it is not installed.
·· Running '/usr/bin/git name-rev --name-only --exclude=remotes/* --no-undefined 43360341b56823ccc2459a0cf9cc709c2b30283f'
   OUTPUT -------->
   shiny-new-fix
·· Building 43360341 <shiny-new-fix> for conda-py3.8-Cython-MDAnalysisTests-mock-nose-numpy-pytest-scipy-six
·· Running '/home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/bin/python -mpip wheel --no-deps --no-index -w /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/asv-build-cache/43360341b56823ccc2459a0cf9cc709c2b30283f /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/project/package'
   OUTPUT -------->
   Processing /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/project/package
     Preparing metadata (pyproject.toml): started
     Preparing metadata (pyproject.toml): finished with status 'error'
     error: subprocess-exited-with-error
     
     × Preparing metadata (pyproject.toml) did not run successfully.
     │ exit code: 255
     ╰─> [2 lines of output]
         MDAnalysis requires Python 3.9+. Python 3.8 detected
         Please upgrade your version of Python.
         [end of output]
   
     note: This error originates from a subprocess, and is likely not a problem with pip.
   error: metadata-generation-failed

   
   × Encountered error while generating package metadata.
   ╰─> See above for output.
   
   note: This is an issue with the package mentioned above, not pip.
   hint: See above for details.

·· Error running /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/bin/python -mpip wheel --no-deps --no-index -w /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/asv-build-cache/43360341b56823ccc2459a0cf9cc709c2b30283f /home/sam/mdanalysis/benchmarks/env/09b6664d368df460776d8e73bd3171a0/project/package (exit status 1)
·· Failed to build the project and import the benchmark suite.`
```

**Comment by Sumit112192** [comment]
I changed the Python version in asv.conf.json to 3.9 to run the benchmark test. 
Attached are the benchmark results:
[benchmarkOutput.log](https://github.com/MDAnalysis/mdanalysis/files/13393297/benchmarkOutput.log)

Can you explain what the results are? @hmacdope @RMeli 
They are in --verbose mode, so the output is too large. Should I run it without verbose, so it might be easier to go through?

**Comment by hmacdope** [comment]
There should be a little summary at the end that compares performance of the two runs. Try again without verbose?

**Comment by Sumit112192** [comment]
@hmacdope, I tried without verbose. But it was still without any summary.

**Comment by hmacdope** [comment]
@Sumit112192 I will run benchmarks for you, sorry for delay.

**Review by tylerjereddy — COMMENTED** [review]
The diff looks "ok" to me. The benchmark suite is currently broken because of gh-3519 when I check locally, and it looks like several other issues may be present as well.

**Comment by tylerjereddy** [comment]
FWIW, I'd probably just vote to merge it. The benchmark suite is a bit of a mess at the moment it seems. I'll see if I can re-run with the failing benchmarks skipped using a regex, but it is annoying to do.

**Comment by hmacdope** [comment]
Similar thing to report, no performance regressions observed on my run. Benchmarks not all running at @tylerjereddy observed, but I am also pretty happy with the diff. I will approve.

File: package/CHANGELOG [hunk 1]
```diff
@@ -303,6 +303,7 @@ Fixes
   * fixes guessed_attributes and read_attributes methods of the Topology class, as 
     those methods were giving unexpected results for some topology attributes
     (e.g. bonds, angles) (PR #3779).
+  * Updated cimport for numpy to maintain cimport consistency (Issue #3908).
```
Comment (hmacdope) [inline package/CHANGELOG]: Newest first.

**Review by hmacdope — CHANGES_REQUESTED** [review]
Just changelog.

**Comment by hmacdope** [comment]
Also @Sumit112192 is this your first contribution? If so add yourself to AUTHORS.

**Comment by Sumit112192** [comment]
@hmacdope I force-pushed the latest commit in which I updated the CHANGELOG and AUTHORS files. Is there any issue with the force commit?

**Comment by Sumit112192** [comment]
@hmacdope I read that force pushing can be dangerous as it deletes other commit history. How can I delete my above force push? Or should I create another pull request using normal push?

**Comment by hmacdope** [comment]
> @hmacdope I read that force pushing can be dangerous as it deletes other commit history. How can I delete my above force push? Or should I create another pull request using normal push?

The diff looks fine to me. no easy way to fix a force push but it's OK, your PR will all be squashed into one commit. No need to stress. We can continue here on this PR

**Comment by Sumit112192** [comment]
@hmacdope relieved to hear that it is not a big issue. Also, I would like to work on the benchmark issue thing. How to get started on it? Any literature, I can follow to learn more about it?

**Comment by Sumit112192** [comment]
That closed thing happened by mistake. Sorry for that.

File: package/CHANGELOG [hunk 2]
```diff
@@ -28,6 +28,7 @@ Fixes
     RMSD of None
   * Fixes hydrogenbonds tutorial path to point to hbonds (Issue #4285, PR #4286)
   * Fix atom charge reading in PDBQT parser (Issue #4282, PR #4283)
+  * Updated cimport for numpy to maintain cimport consistency (Issue #3908).
```
Comment (hmacdope) [inline package/CHANGELOG]: You need to move this to the top of the `Fixes` section as we normally do newest first.

**Review by hmacdope — CHANGES_REQUESTED** [review]
Just the newest first change, might have got lost in the force push, but if you could just make the change again should be good to go.

File: package/CHANGELOG (hunk 2 above)
Comment (Sumit112192) [reply package/CHANGELOG]: Done.

**Review by hmacdope — APPROVED** [review]
Great work @Sumit112192! Super happy that you managed to improve our `cimport` consistency. Has been needed for a while. :) I will wait for @tylerjereddy to also approve, then good to roll.

**Comment by tylerjereddy** [comment]
I squash-merged per MDA convention (I think), thanks!

Only two files had changed since my positive review, and they were just author/changelog things.

I doubt there will be any performance surprises, but hopefully our benchmarks are in better shape soon.

**Pull request merged**